dcos/url_builder_test.go

```diff
@@ -0,0 +1,62 @@
+package dcos
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAgentNoTLS(t *testing.T) {
```

Comment:
I think this file would benefit a lot from table driven tests. Not sure how you feel about them, but to me, it is much easier to read, and much more succinct.